### DISCUSSION TITLE
Using Enums in where rules error: cannot find symbol

### DISCUSSION THREAD

**Discussion by devent** [evrete/evrete#99] [General]
Hi. I start a discussion if this is a bug or just my misunderstanding. `NeighboringDir` is a Java Enum. if I try to use the Enum in the rule `@Where(value = { "$f.block.isNeighborsFilled(NeighboringDir.U)" })` I get the error `error: cannot find symbol`

Minimal example:

```
import com.anrisoftware.dwarfhustle.model.api.objects.NeighboringDir;

public class TerrainCreateRules {
    @Rule(salience = 10)
    @Where(value = { "$f.block.isNeighborsFilled(NeighboringDir.U)" })
    public void block_is_hidden(TerrainFact $f, RhsContext ctx) {
        $f.block.setHidden(true);
    }
}
```

```
WARNING: JavaSourceObject:org.evrete.spi.minimal.compiled.Rule1:19: error: cannot find symbol
    return var0.isNeighborsFilled(NeighboringDir.U);
                                  ^
  symbol:   variable NeighboringDir
  location: class org.evrete.spi.minimal.compiled.Rule1
```

```
WARNING: 
In source:
package org.evrete.spi.minimal.compiled;

public final class Rule1 extends com.anrisoftware.dwarfhustle.model.knowledge.evrete.TerrainCreateRules {
  public static final java.lang.invoke.MethodHandle CONDITION1_HANDLE;

  static {
    try {
      CONDITION1_HANDLE = java.lang.invoke.MethodHandles.lookup().findStatic(Rule1.class, "condition1", java.lang.invoke.MethodType.methodType(boolean.class, org.evrete.api.IntToValue.class));
    } catch (Exception e) {
      throw new IllegalStateException(e);
    }
  }

  public static boolean condition1(org.evrete.api.IntToValue values) {
    return condition1Inner((com.anrisoftware.dwarfhustle.model.api.objects.MapBlock) values.apply(0));
  }

  private static boolean condition1Inner(com.anrisoftware.dwarfhustle.model.api.objects.MapBlock var0) {
    return var0.isNeighborsFilled(NeighboringDir.U);
  }


}
```

If this is really a bug, I can convert it to a bug report.

**Comment by andbi** [comment]
The enum is probably not being actually used by your ruleset class (`TerrainCreateRules`), and your `import com.anrisoftware.dwarfhustle.model.api.objects.NeighboringDir` statement gets discarded by Java compiler. 

Please try adding an explicit import when initializing the `KnowledgeService `:

```java
Configuration conf = new Configuration();
conf.addImport(NeighboringDir.class);

KnowledgeService service = new KnowledgeService(conf);
```

Alternatively, you can use the `@MethodPredicate` annotation and express your condition through a Java method. For more details, see the sections on [@MethodPredicate](https://www.evrete.org/docs#ajr-ref-method-predicate) and [@Where](https://www.evrete.org/docs#ajr-method-where) in the documentation.

**Reply by devent** [reply]
Thank you that works. But I also like to statically import the Enum constants. There is no way to statically import the constants? There is no `conf.addStaticImport`

```
@Where(value = { "$f.block.isNeighborsFilled(U)" })
```

**Comment by andbi** [comment]
Something like `.addImport("static " + NeighboringDir.class.getName() + "." + NeighboringDir.U.name());` should do the trick. If compilation fails, you'll be able to see how to adjust the argument.

`.addStaticImport` is a good idea, noted.

**Reply by devent** [reply]
That works, great thank you.
